src/test/java/spoon/test/template/SubstitutionTest.java

```diff
@@ -114,4 +116,36 @@ public SingleConstructorTemplate() { }
         public void statement() {
         }
     }
+
+    @Test
+    public void testInsertAllSuperInterfaces() {
+        // contract: Substitution.insertAllSuperInterfaces inserts the only superInterface from a single interface implementing template into the target class
+
+        // arrange
+        Launcher spoon = new Launcher();
+        spoon.addTemplateResource(new FileSystemFile("./src/test/java/spoon/test/template/SubstitutionTest.java"));
+
+        spoon.buildModel();
+        Factory factory = spoon.getFactory();
```

Comment:
I'm starting to see a pattern here: these four lines occur in every single test case of this class, but only the factory is ever used. I suggest extracting this into a helper method like so:

```java
    private static Factory createFactoryWithTemplates() {
        Launcher spoon = new Launcher();
        spoon.addTemplateResource(new FileSystemFile("./src/test/java/spoon/test/template/SubstitutionTest.java"));
        spoon.buildModel();
        return spoon.getFactory();
    }
```

And then use that in the arrange part of the tests. In the interest of not piling up PRs that depend on each other, I think it's fine to do that refactoring in this PR. What do you think?

Reply:
Yeah, this makes sense.

I have one question, where should I place this new method? below all the tests I guess?

Reply:
Yeah, I would place it at the bottom.

Reply:
@slarse I think it's fixed now

Reply:
Yep, looks good!